server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const ROOT = __dirname;
+
+// Apply colors to log levels
+const colorizer = colorize({ all: true });
+
+const myFormat_console = printf(({ level, message, timestamp }) => {
+  return `${timestamp} ${level}: ${message}`;
+});
+
+const logger = new winston.createLogger({
+  level: 'info',
+  defaultMeta: { service: 'user-service' },
+  transports: [
+    new winston.transports.Console({
+      format: combine(colorizer, timestamp(), myFormat_console),
+    }),
+    new winston.transports.File({
+      filename: 'server.log',
```

Comment:
The filename for the log file should be defined as a constant to avoid hardcoding. This improves maintainability. 
```diff
-      filename: 'server.log',
+      filename: LOG_FILE,
```
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*